src/targets/gpu/compile_hip_code_object.cpp

```diff
@@ -140,8 +140,8 @@ compute_global_for(context& ctx, std::size_t n, std::size_t over)
         std::size_t max_blocks   = max_global / local;
         std::size_t nglobal      = std::min(max_blocks * over, groups) * local;
 #ifdef MIGRAPHX_USE_HIPRTC
-        if(enabled(MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS{}))
-            num_elements = ((num_elements + local - 1) / local) * local;
+        // hip compiler assumes global workitems is always multiple of local workitems 
+        num_elements = ((num_elements + local - 1) / local) * local;
```

Comment:
This can affect performance, we should only do this with the workarounds enabled. This should be reported as a bug to hiprtc.

Reply:
It was reported as a bug and they came back saying it assumes and requires global workitems multiple of local.

You can check SWDEV-398603.

Reply:
If i dump assembly with hipclang it has this metadata:
 .uniform_work_group_size: 1

hip runtime would start throwing errors if it is not uniform workgroup size. It does show any errors right now, apparently there another bug SWDEV-403995

Reply:
https://reviews.llvm.org/D155213

HIP would allow non-uniform WG sizes in future but for now by default it is not allowed.